client/src/app/site/topics/modules/topic-poll/components/topic-poll-detail-content/topic-poll-detail-content.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Subscription } from 'rxjs';
+
+import { TopicPollService } from '../../services/topic-poll.service';
+
+@Component({
+  selector: `os-topic-poll-detail-content`,
+  templateUrl: `./topic-poll-detail-content.component.html`,
+  styleUrls: [`./topic-poll-detail-content.component.scss`],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TopicPollDetailContentComponent extends BaseComponent implements OnDestroy {
+    private _poll: PollData;
+
+    public get chartData(): ChartData {
+        return this._chartData;
+    }
+
+    public get tableData(): PollTableData[] {
+        return this._tableData;
+    }
+
+    @Input()
+    public set poll(pollData: PollData) {
+        this._poll = pollData;
+        this.setupTableData();
+        this.cd.markForCheck();
+    }
+
+    public get poll(): PollData {
+        return this._poll;
+    }
+
+    @Input()
+    public iconSize: 'large' | 'gigantic' = `large`;
+
+    private get state(): PollState {
+        return this.poll.state;
+    }
+
+    public get hasResults(): boolean {
+        return this.isFinished || this.isPublished;
+    }
+
+    public get isFinished(): boolean {
+        return this.state === PollState.Finished;
+    }
+
+    public get isPublished(): boolean {
+        return this.state === PollState.Published;
+    }
+
+    public get canSeeResults(): boolean {
+        return this.operator.hasPerms(this.permission.pollCanManage) || this.isPublished;
+    }
+
+    public get showChart(): boolean {
+        return this.pollService.showChart(this.poll);
+    }
+
+    private _tableDataSubscription: Subscription | null = null;
+    private _tableData: PollTableData[] = [];
+    private _chartData: ChartData = null;
+
+    public constructor(
+        componentServiceCollector: ComponentServiceCollector,
+        protected translate: TranslateService,
+        private pollService: TopicPollService,
+        private cd: ChangeDetectorRef,
+        private operator: OperatorService
+    ) {
+        super(componentServiceCollector, translate);
+    }
+
+    public ngOnDestroy(): void {
+        super.ngOnDestroy();
+        this.cleanup();
+    }
+
+    private setupTableData(): void {
+        this.cleanup();
+        this._tableDataSubscription = this.pollService.generateTableDataAsObservable(this.poll).subscribe(tableData => {
+            this._tableData = tableData;
+            this.setChartData();
+            this.cd.markForCheck();
+        });
+    }
+
+    private setChartData(): void {
+        this._chartData = this.pollService.generateChartData(this.poll);
+    }
+
+    private cleanup(): void {
+        if (this._tableDataSubscription) {
+            this._tableDataSubscription.unsubscribe();
+            this._tableDataSubscription = null;
+        }
+    }
+
+    public get chartColors(): [{backgroundColor: string[]}] {
+        console.log(this.poll.options);
+        return [this.generateChartColors(this.poll.options.length)];
+    }
```

Comment:
Either make a real function (`getChartColors()`) or put it to the other member variables and accessors.